lib/playground.dart

```diff
@@ -353,12 +353,18 @@ class Playground {
 
     _clearOutput();
 
+    Stopwatch compilationTimer = new Stopwatch()..start();
+
     var input = new SourceRequest()..source = context.dartSource;
     dartServices.compile(input).timeout(longServiceCallTimeout).then(
         (CompileResponse response) {
+      ga.sendTiming(
+          "action-perf", "compilation-e2e", compilationTimer.elapsedMilliseconds);
+
       return executionService.execute(
           _context.htmlSource, _context.cssSource, response.result);
     }).catchError((e) {
+      ga.sendException(e.runtimeType.toString());
```

Comment:
You can also to `${e.runtimeType}`, which I'm in the habit of doing instead of toString() in many situations just to avoid the chance of an NPE.Which, I don't think is possible at this point.